src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,46 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+
+    private static final int MAX_ARRAY_SIZE = 10;
+    private final K[] keys;
+    private final V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_ARRAY_SIZE];
+        values = (V[]) new Object[MAX_ARRAY_SIZE];
+    }
+
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < MAX_ARRAY_SIZE; i++) {
+            if (keys[i] == key || values[i] == null || (keys[i] != null && keys[i].equals(key))) {
+                keys[i] = key;
+                values[i] = value;
+                break;
+            }
+        }
     }
 
     @Override
     public V get(K key) {
+        for (int i = 0; i < MAX_ARRAY_SIZE; i++) {
```

Comment:
```suggestion
        int index = indexOf(key);
        return index == -1 ? null : values[index];
```

create indexOf(K key) method